store.py

```diff
@@ -216,46 +409,200 @@ def accept(self, customer: Customer) -> bool:
+            return True
+        return False
 
     def start_checkout(self) -> int:
         """Checkout the next customer in this CheckoutLine.
-
         Return the time it will take to checkout the next customer.
+        Assume that there is a customer in line when this is called.
         """
+        raise NotImplementedError
 
     def complete_checkout(self) -> bool:
         """Finish the checkout for this CheckoutLine.
 
         Return whether there are any remaining customers in the line.
+        Customer is in the queue until they are done checking out.
+        >>>
         """
+        self.queue.pop(0)
+        return len(self.queue) >= 1
 
     def close(self) -> List[Customer]:
         """Close this line.
 
         Return a list of all customers that need to be moved to another line.
+        The last person in line will join another queue (first) as the line
+        closes, and the rest of the people will join each second after.
         """
+        i = 0
+        result = []
+        reverse_queue = self.queue[::-1]
+        while i != (len(self.queue) - 1):
+            to_move = reverse_queue.pop(i)
+            result.append(to_move)
+            i += 1
+        return result
+
 
 
-# TODO: implement the following subclasses of CheckoutLine
-# Only implement those methods that cannot be used exactly as inherited
-# from the superclass.
-# You may add private attributes and helper methods, but do not change the
-# public interface of the subclasses.
-# Write docstrings for all methods you write, and document your attributes
-# in the class docstring.
 
 class RegularLine(CheckoutLine):
-    """A regular CheckoutLine."""
+    """A regular CheckoutLine.
+
+    Any customer can join the line, if there is room.
+    The time required to checkout is equal to the total time required
+    for items the customer has.
+
+    === Attributes ===
+    capacity: The number of customers allowed in this CheckoutLine.
+    is_open: True iff the line is open.
+    queue: Customers in this line in FIFO order.
+
+    === Representation Invariants ===
+    - Each customer in this line has not been checked out yet.
+    - The number of customers is less than or equal to capacity.
+    """
+
+    capacity: int
+    is_open: bool
+    queue: List[Customer]
+
+    def __init__(self, capacity: int, is_open: True, queue: []) -> None:
+        """Initialize an open and empty RegularLine .
+
+        >>> line = RegularLine(1, True, []) #Rewrite this doctest
+        >>> line.capacity
+        1
+        >>> line.is_open
+        True
+        >>> line.queue
+        []
+        """
+        super().__init__(capacity, is_open, queue)
+
+    def start_checkout(self) -> int:
+        """Checkout the next customer in this CheckoutLine.
+
+        Return the time it will take to checkout the next customer.
+        Assume that there is a customer in line when this is called.
+        """
+        total_time = 0
+        customer = self.queue[0]
```

Comment:
are you sure this assumption is correct ?  
I'd catch it and returned 0 instead